### PR TITLE
fixes documentation for returned hash in cache.fetch_multi

### PR DESCRIPTION
**Pull request by mraidel**

### CODE REVIEW

**Comment by schneems** [comment]
This is a confusing example in the first place, it's not clear what's in the block is a fallback value and that "boom" is expected to not be in the cache. Maybe put an explicit `cache.read("boom")` in there to show it returns nil. We could also make the block return a more helpful result maybe `{|k| "Fallback value for key: #{k}"}`

**Comment by mraidel** [comment]
@schneems added a more clear example

File: activesupport/lib/active_support/cache.rb [hunk 1]
```diff
@@ -352,9 +352,14 @@ def read_multi(*names)
       #
       # Returns a hash with the data for each of the names. For example:
       #
-      #   cache.write("bim", "bam")
-      #   cache.fetch_multi("bim", "boom") { |key| key * 2 }
-      #   # => { "bam" => "bam", "boom" => "boomboom" }
+      #   cache.write("new_key", "value")
+      #   cache.fetch_multi("new_key", "unknown_key") do |key|
```
Comment (kaspth) [inline activesupport/lib/active_support/cache.rb]: Calling it "new_key" makes it more confusing here. As the key is known it's not new.

Comment (kaspth) [inline activesupport/lib/active_support/cache.rb]: Calling it "new_key" makes it more confusing here. As the key is known it's not new.

File: activesupport/lib/active_support/cache.rb [hunk 2]
```diff
@@ -352,9 +352,14 @@ def read_multi(*names)
...
+      #     "value for: #{key}"
+      #   end
+      #   # => {
+      #   #   "new_key" => "value",
```
Comment (kaspth) [inline activesupport/lib/active_support/cache.rb]: Same goes for using "value" here. Its harder to read when the fallback is "value for..."

**Comment by mraidel** [comment]
@kaspth what about:
```diff
--- a/activesupport/lib/active_support/cache.rb
+++ b/activesupport/lib/active_support/cache.rb
@@ -352,13 +352,13 @@ module ActiveSupport
       #
       # Returns a hash with the data for each of the names. For example:
       #
-      #   cache.write("new_key", "value")
-      #   cache.fetch_multi("new_key", "unknown_key") do |key|
-      #     "value for: #{key}"
+      #   cache.write("foo", "bar")
+      #   cache.fetch_multi("foo", "unknown_key") do |key|
+      #     "new value for: #{key}"
       #   end
       #   # => {
-      #   #   "new_key" => "value",
-      #   #   "unknown_key" => "value for: unknown_key"
+      #   #   "foo" => "bar",
+      #   #   "unknown_key" => "new value for: unknown_key"
       #   # }
       #
       def fetch_multi(*names)
```

**Comment by kaspth** [comment]
Then it would be just as good to use the bim and bam names from before, so the diff is smaller.

**Comment by mraidel** [comment]
@kaspth there we go :-). Should I rebase to one commit?

**Comment by kaspth** [comment]
Yes, only one commit :wink:

I would still like to have the word fallback (instead of new) in the fetch block as @schneems suggested. "New" sounds like it might have overwritten an old value for `unknown_key`.

**Comment by mraidel** [comment]
@kaspth done!

**Comment by kaspth** [comment]
Reading through it, it's still not super clear to me. Let's have @schneems look it over one more time :smile:

**Comment by schneems** [comment]
This is looking much better. One last thing and I think we're good to go. Can you change the hash so that the curly's are not on a new line

Here's a good example https://github.com/rails/rails/blob/c2108926a4638b5373f846c2d6165bf28108b9cb/activemodel/lib/active_model/serializers/json.rb#L64-L66

Make sure to put a space after between the curly brackets and the inside contents.

Appreciate your help :smile_cat:

**Comment by mraidel** [comment]
@schneems pushed!

**Comment by schneems** [comment]
This looks great :heart:

**Comment by schneems** [comment]
Oh and btw for documentation commits in the future you can add "ci skip" to the commit message if the PR doesn't touch any code (only comments and docs) http://docs.travis-ci.com/user/how-to-skip-a-build/. That helps us not bog travis down running code it doesn't need to. No worries on this one, but in the future...now you know! Thanks again!

**Comment by kaspth** [comment]
@mraidel Thanks :heart:

**Pull request merged**